tests/test_registrations/test_models.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+'''Unit tests for models and their factories.'''
+from nose.tools import *  # noqa (PEP8 asserts)
+import mock
+import unittest
+
+from website.models import MetaSchema, DraftRegistrationApproval
+
+from tests.factories import (
+    UserFactory, ApiOAuth2ApplicationFactory, NodeFactory, PointerFactory,
+    ProjectFactory, NodeLogFactory, WatchConfigFactory,
+    NodeWikiFactory, RegistrationFactory, UnregUserFactory,
+    ProjectWithAddonFactory, UnconfirmedUserFactory, CommentFactory, PrivateLinkFactory,
+    AuthUserFactory, DashboardFactory, FolderFactory,
+    NodeLicenseRecordFactory, DraftRegistrationFactory
+)
+from tests.test_registrations.base import RegistrationsTestBase
+
+
+class TestDraftRegistrations(RegistrationsTestBase):
+
+    def test_factory(self):
+        draft = DraftRegistrationFactory()
+        assert_is_not_none(draft.branched_from)
+        assert_is_not_none(draft.initiator)
+        assert_is_not_none(draft.registration_schema)
+
+        user = AuthUserFactory()
+        draft = DraftRegistrationFactory(initiator=user)
+        assert_equal(draft.initiator, user)
+
+        node = ProjectFactory()
+        draft = DraftRegistrationFactory(branched_from=node)
+        assert_equal(draft.branched_from, node)
+        assert_equal(draft.initiator, node.creator)
+
+        schema = MetaSchema.find()[1]
+        data = {'some': 'data'}
+        draft = DraftRegistrationFactory(registration_schema=schema, registration_metadata=data)
+        assert_equal(draft.registration_schema, schema)
+        assert_equal(draft.registration_metadata, data)
+
+    @mock.patch('website.project.model.Node.register_node')
+    def test_register(self, mock_register_node):
+
+        self.draft.register(self.auth)
+        mock_register_node.assert_called_with(
+            schema=self.draft.registration_schema,
+            auth=self.auth,
+            data=self.draft.registration_metadata,
+        )
+
+    def test_update_metadata_tracks_changes(self):
+        self.draft.registration_metadata = {
+            'foo': {
+                'value': 'bar',
+            },
+            'a': {
+                'value': 1,
+            },
+            'b': {
+                'value': True
+            },
+        }
+        changes =  self.draft.update_metadata({
+            'foo': {
+                'value': 'foobar',
+            },
+            'a': {
+                'value': 1,
+            },
+            'b': {
+                'value': True,
+            },
+            'c': {
+                'value': 2,
+            },
+        })
+        self.draft.save()
+        for key in ['foo', 'c']:
+            assert_in(key, changes)
+
+    @unittest.skip("TODO(samchrisinger): update this")
```

Comment:
TODO noted. If I understand this correctly, it applies to a feature that won't be usable until the admin tool is ready?

If that's the case, I'm ok with us just making a record, and fixing this in a followup PR.